### ISSUE TITLE
:: brack  autocomplete prosses

### ISSUE THREAD

**Issue by a-moses** [atom-community/autocomplete-snippets#39]
the sign ' : ' brack the autocomplete 

for example

Closure Closure::bind

**Comment by funsterisdafunniest** [comment]
have run into this issue too while trying out the snippets for language-laravel. is there any way to fix this?
after we type ':' the autocomplete think that we're trying to type a completely new snippets and didn't take into consideration about what was written before the ':' character

**Comment by joefitzgerald** [comment]
Can you please post the full snippet text for a snippet that exhibits this behavior?

**Comment by funsterisdafunniest** [comment]
of course!
it's part of the atom package here https://atom.io/packages/laravel
when i try to use the snippet for Route::get snippet, i need to type ro:g for it to work.
but if autocomplete-snippets and autocomplete-plus is on, it consider the g as a new autocomplete keyword / search

**Comment by joefitzgerald** [comment]
Yes, it is an autocomplete-plus bug. https://github.com/Cronos87/atom-laravel/blob/cebcc3174c5d1d8834eebc674708f270749a0c53/snippets/laravel.cson#L24-L27 seems to capture a test case that we can use.

**Comment by joefitzgerald** [comment]
Your Route::get snippet is here: https://github.com/Cronos87/atom-laravel/blob/cebcc3174c5d1d8834eebc674708f270749a0c53/snippets/laravel.cson#L451-L454. I'll use that.

**Comment by funsterisdafunniest** [comment]
hi, i've found out the modification needed to allow autocomplete-plus to recognize snippets with colon characters.
it's these
https://github.com/atom-community/autocomplete-plus/blob/master/lib/autocomplete-manager.coffee#L34-L35

https://github.com/atom-community/autocomplete-plus/blob/master/lib/fuzzy-provider.coffee#L11

should i do a pull request? since this modifies the fuzzy provider which, according to the wiki, is already deprecated

**Comment by joefitzgerald** [comment]
@benogle is it valid for a snippet's groups to be nested like this: `Route::input(${1:'${2:name}'})$3` (i.e. group 2 is nested inside of 1). I have replicated a failure in our autocomplete-snippet parsing if the groups are nested. The double `::` doesn't cause any issues.
